modules/schema/src/main/java/org/apache/ignite/internal/schema/BinaryTupleComparatorUtils.java

```diff
@@ -122,6 +122,12 @@ static int compareAsString(BinaryTupleReader tuple, int colIndex, String cmp, bo
         byte[] bytes = tuple.bytesValue(begin, begin + trimmedSize);
         char[] cmpArray = cmp.toCharArray();
 
+        if (bytes.length < trimmedSize) {
+            assert bytes.length == trimmedSize - 1 : "Only one first byte can have a special value.";
+
+            fullSrtLength--;
```

Comment:
What's "srt"?

Reply:
It is a reduction of the full size of string value in bytes.

Reply:
I don't understand the variable name, shouldn't it be `fullStrLength`?

Reply:
Renamed to fullStrLength